internal/controllers/parser.go

```diff
@@ -83,6 +84,7 @@ func UpdateConfigFromAPIOpts(
 	httpConnectionManagerBuilder *config.HCMBuilder,
 	clBuilder *cloudentity.Builder,
 	name string,
+	kubernetesClient client.Client,
```

Comment:
this function is becoming uglier. We must do something about it

Reply:
I agree.